source/common/access_log/access_log_formatter.cc

```diff
@@ -912,22 +912,32 @@ FilterStateFormatter::FilterStateFormatter(const std::string& key,
                                            absl::optional<size_t> max_length)
     : key_(key), max_length_(max_length) {}
 
-ProtobufTypes::MessagePtr
+const Envoy::StreamInfo::FilterState::Object*
 FilterStateFormatter::filterState(const StreamInfo::StreamInfo& stream_info) const {
   const StreamInfo::FilterState& filter_state = stream_info.filterState();
   if (!filter_state.hasDataWithName(key_)) {
     return nullptr;
   }
-
-  const auto& object = filter_state.getDataReadOnly<StreamInfo::FilterState::Object>(key_);
-  return object.serializeAsProto();
+  return &filter_state.getDataReadOnly<StreamInfo::FilterState::Object>(key_);
 }
 
 std::string FilterStateFormatter::format(const Http::RequestHeaderMap&,
                                          const Http::ResponseHeaderMap&,
                                          const Http::ResponseTrailerMap&,
                                          const StreamInfo::StreamInfo& stream_info) const {
-  ProtobufTypes::MessagePtr proto = filterState(stream_info);
+  const Envoy::StreamInfo::FilterState::Object* state = filterState(stream_info);
+  if (!state) {
+    return UnspecifiedValueString;
+  }
+
+  // Prefer serializeAsString for better performance
```

Comment:
Should we have explicit formatters (and format specifier) for raw string vs. JSON?

Reply:
> Should we have explicit formatters (and format specifier) for raw string vs. JSON?

That's better. It is up to the user to decide which serialization interface to use. I will get it done as soon as possible.